### ISSUE TITLE
Allow setting minimum length of words

### ISSUE THREAD

**Issue by yiminzme** [yiminzme/Multi-Highlight#28]
Users normally don't search letters (e.g., "a", "b", etc), while typing a letter can cause lag on a lengthy page when "Instant Search" mode is enabled.
For example, type in "a" on [example](https://patents.google.com/patent/US5877759A/en?oq=google+patent).

We can set a threshold (e.g., 2), and filter out any word with length less than the threshold.

**Comment by TroyDanielFZ** [comment]
It's very easy. In `keywordsFromStr`, change `filter(i=>i)` into `filter(i=>i.length>=2)` should work fine.

**Comment by yiminzme** [comment]
👍 Ah, I see. Anyone who is available can implement it, will merge the commit.

**Comment by yiminzme** [comment]
Then maybe we can replace the "Instant Search" with this more flexible setting. When threshold=1, it is equivalent to "Instant Search".
What do you think of this feature @TroyDanielFZ

**Comment by TroyDanielFZ** [comment]
> threshold=1

Do you mean, highlight in a `setTimeout` function? To be more specific, if `delayTime=0`, apply highlighting immediately. If `delaytime>0`, then appling highlighting after `delayTime` (in milliseconds) after the last keyboard event?

If so, I approve of it.

**Comment by yiminzme** [comment]
No, you mentioned that we can use `filter(i=>i.length>=2)`. The 2 is the threshold for the minimum length of words.
So we can define the threshold as a variable, e..g. `var min_length_threshold`. Set it to 2 by default. So `filter(i=>i.length>=min_length_threshold)`. And allow users to change `min_length_threshold`.
(When `min_length_threshold` is 1, it is equivalent to enabling the "Instant Search" feature.)

**Comment by yiminzme** [comment]
For reference, the "Instant Search" features search word as user type. 
When it is disabled, Multi Highlight only searches a word when a delimiter is placed after the word.

**Comment by TroyDanielFZ** [comment]
> So `filter(i=>i.length>=min_length_threshold)`. And allow users to change `min_length_threshold`.

Yep, I get it. I approve of it.

Please give me some time to think, which options do we need. And right now, there are so many options, we should group the options into different groups, for instance, (1) Color settings; (2) Match options; (3) Startup options. This example is not good, just to illustrate what I mean.

> Multi Highlight only searches a word when a delimiter is placed after the word.

I didn't use this feature before. I tick the "Instant Search" everytime.

--- 

Btw, a suggestion. We should split the document, for example, to the root folder `doc`, to minimize the installed extenstion size. And the doc can be online available with git wiki, I guess.

**Comment by yiminzme** [comment]
> Btw, a suggestion. We should split the document, for example, to the root folder doc, to minimize the installed extenstion size. And the doc can be online available with git wiki, I guess.

I think so. Good idea. 👍
